### ISSUE TITLE
Oscar need to precompile every time load it.

### ISSUE THREAD

**Issue by PhyX-Meow** [oscar-system/Oscar.jl#2203]
**Describe the bug**
It somehow requires precompile every time I start a new julia session and `using Oscar`. It seem no functionality is affected.

**System:**
```
julia> Oscar.versioninfo(full=true)
OSCAR version 0.12.0-DEV
  combining:
    AbstractAlgebra.jl   v0.29.2
    GAP.jl               v0.9.4
    Hecke.jl             v0.18.7
    Nemo.jl              v0.33.7
    Polymake.jl          v0.9.1 - #master, 3c04e0ad -- 2023-03-06 10:35:24 +0100
    Singular.jl          v0.18.1
  building on:
    Antic_jll               v0.201.500+0
    Arb_jll                 v200.2300.0+0
    Calcium_jll             v0.401.100+0
    FLINT_jll               v200.900.5+0
    GAP_jll                 v400.1200.200+0
    Singular_jll            v403.101.500+0
    libpolymake_julia_jll   v0.9.0+0
    libsingular_julia_jll   v0.29.0+0
    polymake_jll            v400.900.0+0
See `]st -m` for a full list of dependencies.

Julia Version 1.9.0-rc2
Commit 72aec423c2a (2023-04-01 10:41 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 16 × AMD Ryzen 7 3700X 8-Core Processor
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, znver2)
  Threads: 1 on 16 virtual cores
Environment:
  LD_LIBRARY_PATH = .:../lib:./lib
  JULIA_PKG_SERVER = https://mirrors.tuna.tsinghua.edu.cn/julia
  LD_PRELOAD = /usr/lib/libstdc++.so.6
Official https://julialang.org/ release
```

**Comment by thofma** [comment]
Could you post the output of your terminal, which exhibits the problem. E.g. start julia, load Oscar, exit julia, start julia, load Oscar.

**Comment by benlorenz** [comment]
We were relying on some undocumented behavior of `include_dependency` by passing it a directory and not a regular file. This was working fine in julia 1.6 and 1.8 but changed (probably not on purpose) in master and 1.9. Now julia is rejecting the cache file with a slightly misleading message (with `JULIA_DEBUG=all`):
```
┌ Debug: Rejecting stale cache file <...>/aJ3Pg_wOMvI.ji because file <...>/Oscar.jl/experimental does not exist
```

I have opened a PR for julia to get this working again and documented. If this is not accepted we need to adjust our code and list all experimental modules explicitly instead.

cc: @lkastner

**Comment by PhyX-Meow** [comment]
Thank you! I'd like to come back using 1.8 until this is fixed.

**Comment by PhyX-Meow** [comment]
Fixed in 1.9.0-rc3, closing this.
